### PR TITLE
feat(pipeline): Augment latest Pipelineruns to Pipeline List view

### PR DESCRIPTION
**Pull request by abhinandan13jan**
Pipelinerun details are added to Pipeline List view and is sortable
story: https://jira.coreos.com/browse/ODC-652

Screenshot from 2019-05-21 16-04-00

### CODE REVIEW

File: frontend/public/extend/devconsole/components/pipelines/PipelineAugmentRuns.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable no-undef  */
+import * as React from 'react';
+import { List } from '../../../../components/factory';
+import PipelineHeader from './PipelineHeader';
+import PipelineRow from './PipelineRow';
+import { PipelineListProps } from './PipelineList';
+import { augmentRunsToData } from '../../utils/pipeline-augment';
+
+export interface PipelineAugmentRunsProps extends PipelineListProps {
+  propsReferenceForRuns?: any;
```
Comment (divyanshiGupta) [inline frontend/public/extend/devconsole/components/pipelines/PipelineAugmentRuns.tsx]: Can you add a type here?

File: frontend/public/extend/devconsole/components/pipelines/PipelineAugmentRuns.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+}
+
+const PipelineAugmentRuns = (props: PipelineAugmentRunsProps) => {
```
Comment (divyanshiGupta) [inline frontend/public/extend/devconsole/components/pipelines/PipelineAugmentRuns.tsx]:
```suggestion
const PipelineAugmentRuns: React.FC<PipelineAugmentRunProps> = (props: PipelineAugmentRunsProps) => {
```

File: frontend/public/extend/devconsole/components/pipelines/PipelineList.tsx [hunk 3]
```diff
@@ -1,9 +1,25 @@
...
 
-export const PipelineList = (props) => (
-  <List {...props} Header={PipelineHeader} Row={PipelineRow} />
-);
+/* eslint-disable no-undef */
+export interface PipelineListProps {
+  data?: PropPipelineData[];
+}
+
+const PipelineList = (props: PipelineListProps) => {
```
Comment (divyanshiGupta) [inline frontend/public/extend/devconsole/components/pipelines/PipelineList.tsx]:
```suggestion
const PipelineList: React.FC<PipelineListProps> = (props: PipelineListProps) => {
```

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

File: frontend/public/extend/devconsole/components/pipelines/PipelineAugmentRuns.tsx (hunk 1 above)
Comment (abhinandan13jan) [reply frontend/public/extend/devconsole/components/pipelines/PipelineAugmentRuns.tsx]: done

File: frontend/public/extend/devconsole/components/pipelines/PipelineAugmentRuns.tsx (hunk 2 above)
Comment (abhinandan13jan) [reply frontend/public/extend/devconsole/components/pipelines/PipelineAugmentRuns.tsx]: changed

File: frontend/public/extend/devconsole/components/pipelines/PipelineList.tsx (hunk 3 above)
Comment (abhinandan13jan) [reply frontend/public/extend/devconsole/components/pipelines/PipelineList.tsx]: changed

**Comment by abhinandan13jan** [comment]
@divyanshiGupta Thanks for the suggestions. Made the changes, pls approve

File: frontend/public/extend/devconsole/components/pipelines/__tests__/pipeline-augment.spec.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+          name: 'apple2',
+          namespace: 'myproject',
+        },
+      },
+    ],
+  },
+];
+
+// This will be added by Firehose and PipelineList to be passed to PipelineAugmentRuns
+const additionalProps = [
```
Comment (karthikjeeyar) [inline frontend/public/extend/devconsole/components/pipelines/__tests__/pipeline-augment.spec.ts]: move the mock data to separate file

File: frontend/public/extend/devconsole/components/pipelines/PipelineAugmentRuns.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  propsReferenceForRuns?: string[];
+}
+
+const PipelineAugmentRuns: React.FC<PipelineAugmentRunsProps> = (props: PipelineAugmentRunsProps) => {
+  const newProps = Object.assign({}, props);
+  newProps.data = augmentRunsToData(props);
+  return <List {...newProps} Header={PipelineHeader} Row={PipelineRow} />;
+};
+
+export default PipelineAugmentRuns;
```
Comment (karthikjeeyar) [inline frontend/public/extend/devconsole/components/pipelines/PipelineAugmentRuns.tsx:21]: use prettier to format this file

**Review by karthikjeeyar — APPROVED** [review]
LGTM

**Review by divyanshiGupta — APPROVED** [review]

File: frontend/public/extend/devconsole/components/pipelines/PipelineAugmentRuns.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+export interface PipelineAugmentRunsProps extends PipelineListProps {
+  propsReferenceForRuns?: string[];
+}
+
+const PipelineAugmentRuns: React.FC<PipelineAugmentRunsProps> = (
+  props: PipelineAugmentRunsProps,
+) => {
+  const newProps = Object.assign({}, props);
+  newProps.data = augmentRunsToData(props);
+  return <List {...newProps} Header={PipelineHeader} Row={PipelineRow} />;
```
Comment (christianvogt) [inline frontend/public/extend/devconsole/components/pipelines/PipelineAugmentRuns.tsx]:
```suggestion
  return <List {...props} data={augmentRunsToData(props)} Header={PipelineHeader} Row={PipelineRow} />;
```

**Review by christianvogt — APPROVED** [review]

**Pull request merged**